### ISSUE TITLE
Generating an expression from a local file is broken since --subpath was added

### ISSUE THREAD

**Issue by peti** [NixOS/cabal2nix#274]
`cabal2nix` can no longer convert a local Cabal file into a Nix expression:

~~~
$ cabal2nix --sha256=dontworry $(nix-build --no-out-link "<nixpkgs>" -A all-cabal-hashes)/parsec/3.0.0/parsec.cabal
*** parsing cabal file: /nix/store/spi1dcgsx2p4q526brdijkvkaglj1yhm-all-cabal-hashes-53fcf983669a3f0cdfd795fec28ecb40740a64ca-src/parsec/3.0.0/parsec.cabal/.: openBinaryFile: inappropriate type (Not a directory)
cabal2nix: nix-prefetch-zip: createProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
~~~

Previous versions would simply emit a Nix file without trying to run any prefetch commands. According to `git bisect` the commit that introduced that issue is 85a20766f212626b081945118b21c8a1e51a35b6:

    $ git bisect log    
    git bisect start    
    # bad: [d2334f4e68a6a7d5a4a2b62dff0e62f81800a0b2] fromGenericPackageDescription: make sure the new subpath field is initialized    
    git bisect bad d2334f4e68a6a7d5a4a2b62dff0e62f81800a0b2    
    # good: [c75a2a15ca70b83826b925443e43ab41faed8649] Merge pull request #262 from expipiplus1/bench    
    git bisect good c75a2a15ca70b83826b925443e43ab41faed8649    
    # good: [9e55aabdc3d25be66c0c17a01904e557a77be08b] Bump version number to 2.1.1 for upcoming release.    
    git bisect good 9e55aabdc3d25be66c0c17a01904e557a77be08b    
    # good: [7e5a37f3813dda081a941b0b53158d59429e9f23] Merge pull request #269 from hamishmack/gtksouceview3    
    git bisect good 7e5a37f3813dda081a941b0b53158d59429e9f23    
    # good: [f4a9f3a656ca3de93744432c420b9cd1740a2910] package.yaml: update author list    
    git bisect good f4a9f3a656ca3de93744432c420b9cd1740a2910    
    # bad: [ff83d2d1913a1e4580dd0ff438b1833660854cb3] Cosmetic changes to the way postUnpack is rendered.    
    git bisect bad ff83d2d1913a1e4580dd0ff438b1833660854cb3    
    # bad: [85a20766f212626b081945118b21c8a1e51a35b6] Add --subpath to support repos with Cabal files in subdirectories.    
    git bisect bad 85a20766f212626b081945118b21c8a1e51a35b6    
    # first bad commit: [85a20766f212626b081945118b21c8a1e51a35b6] Add --subpath to support repos with Cabal files in subdirectories.    

This issue breaks, among other things, the `callHackage` feature in Nixpkgs.

Ping: @jmitchell

**Comment by domenkozar** [comment]
Let's make sure we have a test for this.

**Comment by jmitchell** [comment]
I've implemented a failing regression test in [this branch](https://github.com/jmitchell/cabal2nix/tree/fix/274-local-cabal). I will investigate the root cause today or tomorrow.

**Comment by jmitchell** [comment]
This is resolved in PR #275, and the regression test is now passing in CI.
